container/spi/src/main/java/org/jboss/arquillian/container/spi/client/protocol/metadata/Servlet.java

```diff
@@ -68,6 +70,7 @@ void setParent(HTTPContext context) {
         if (context.getHost() == null) {
             throw new IllegalArgumentException(context.getClass().getSimpleName() + " host must not be null");
         }
+        this.scheme = context.isSecure() ? HTTPS_SCHEME : HTTP_SCHEME;
```

Comment:
As we are using this logic in two places now I was wondering if that would make sense to add `context.getScheme()` and encapsulate it there. WDYT?

Reply:
We could do that too. I almost went with that approach, but talked myself into the boolean for some reason :)